ts/features/lollipop/utils/fetch.ts

```diff
@@ -0,0 +1,299 @@
+          method,
+          inputUrl,
+          originalUrl
+        };
+        const mainSignatureConfig: SignatureConfig = forgeSignatureConfig(
+          signatureConfigForgeInput,
+          keyInfo,
+          [
+            "Content-Digest",
+            "Content-Type",
+            "x-pagopa-lollipop-original-method",
+            "x-pagopa-lollipop-original-url"
+          ]
+        );
+
+        newInit = addHeader(
+          newInit,
+          "x-pagopa-lollipop-original-method",
+          method
+        );
+        newInit = addHeader(
+          newInit,
+          "x-pagopa-lollipop-original-url",
+          originalUrl
+        );
+        const newInitHeaders =
+          (newInit.headers as Record<string, string>) ?? {};
+        const {
+          signatureBase: mainSignatureBase,
+          signatureInput: mainSignatureInput
+        } = generateSignatureBase(newInitHeaders, mainSignatureConfig, 1);
+
+        const customContentToSignInput: CutsomContentToSignInput = {
+          customContentToSign: lollipopConfig.customContentToSign,
+          keyInfo,
+          keyTag: requestAndKeyInfo.keyTag,
+          signatureConfigForgeInput
+        };
+
+        try {
+          const mainSignValue = await sign(
+            mainSignatureBase,
+            requestAndKeyInfo.keyTag
+          );
+          const customSignResult = await chainSignPromises(
+            customContentToSignPromises(customContentToSignInput)
+          );
+          // Add custom headers
+          customSignResult.forEach(
+            v => (newInit = addHeader(newInit, v.headerName, v.headerValue))
+          );
+          // Prepare custom signature inputs array
+          const customSignatureInputs = customSignResult.map(
+            v => v.signatureInput
+          );
+          // Prepare custom signature array
+          const customSignatures = customSignResult.map(v => v.signature);
+          // Setup signature array
+          const signatures = [`sig1:${mainSignValue}:`, ...customSignatures];
+          // Setup signature input array
+          const signatureInputs = [
+            mainSignatureInput,
+            ...customSignatureInputs
+          ];
+          // Add all to their corresponding headers
+          newInit = addHeader(newInit, "signature", signatures.join(","));
+          newInit = addHeader(
+            newInit,
+            "signature-input",
+            signatureInputs.join(",")
+          );
+          return await timeoutFetch(input, newInit);
+        } catch {
+          return await timeoutFetch(input, init);
+        }
+      }
+      return timeoutFetch(input, init);
+    }
+  );
+};
+
+export const customContentSignatureBases = (
+  customContent: CutsomContentToSignInput
+): Array<CustomContentBaseSignature> => {
+  const contentToSign = customContent.customContentToSign ?? {};
+  const contentToSignKeys = Object.keys(contentToSign);
+
+  return pipe(
+    contentToSignKeys,
```

Comment:
What do you think about this?

```suggestion
  return pipe(
    Object.keys(customContent.customContentToSign ?? {}),
```